actors/runtime/runtime.go

```diff
@@ -87,6 +87,10 @@ type Runtime interface {
 
 	// Starts a new tracing span. The span must be End()ed explicitly, typically with a deferred invocation.
 	StartSpan(name string) TraceSpan
+
+	// ChargeGas charges specified amount of gas for execution.
+	// Should be used before the operation the charge is for.
+	ChargeGas(gas Gas)
```

Comment:
This is a pretty big structure to pass by value all the time. Consider taking a pointer here, and changing the constructor/builder methods to also return/receive pointers.

Reply:
It is 64bytes, so not that big, by being copied on the stack we are quite possibly saving on a lot of heap allocations (as we are passing it to interface it has to be assumed escaped).
64 bytes is also the size of a cache line on most CPU. Meaning that copy should be very fast if it happens.

I also prefer the immutable semantics of how it is built. So unless you are feeling strongly about it, I won't be changing it.

Reply:
It looks to me like 4 int64s + 2 pointers.

Reply:
Slice is 3 uint64s, so 1 pointer, 4 int64 and slice header (3), 8x uint64.